presto-spi/src/main/java/io/prestosql/spi/procedure/Procedure.java

```diff
@@ -43,7 +46,14 @@ public Procedure(String schema, String name, List<Argument> arguments, MethodHan
 
         Set<String> names = new HashSet<>();
         for (Argument argument : arguments) {
-            checkArgument(names.add(argument.getName()), "Duplicate argument name: " + argument.getName());
+            checkArgument(names.add(argument.getName()), format("Duplicate argument name: '%s'", argument.getName()));
+        }
+
+        // check if optional arguments are last on the list of arguments
```

Comment:
Wording can be improved: 

```
        // check if optional arguments are after required ones
```

However, i feel like the comment can be removed, as the code intent is well documented by the exc msg that we throw.